components/CreateOrUpdateRecipeForm.js

```diff
@@ -14,14 +14,13 @@ export default function CreateOrUpdateRecipeForm({
 
   return (
     <>
-      <h1>{formTitle}</h1>
-
-      {errorMessage && (
-        <StyledErrorMessageContainer>
-          <StyledErrorMessage>{errorMessage}</StyledErrorMessage>
-        </StyledErrorMessageContainer>
-      )}
-      <main>
+      <StyledTitle>{formTitle}</StyledTitle>
+
+      <StyledErrorMessage $isVisible={errorMessage.visible}>
+        {errorMessage.text}
+      </StyledErrorMessage>
```

Comment:
I don't think you have to make your errorMessage an Object with two key value pairs, since an empty String is false anyways and as far as I understand your errorMessage.text would be either a Message in a String or an empty String, right? So if your errorMessage state would just be a Sting this should work, too: 
```suggestion
      <StyledErrorMessage $isVisible={errorMessage ? true : false}>
        {errorMessage}
      </StyledErrorMessage>
```

For this you obviously would have to change it at multiple lines in your code but maybe it would shorten it a bit. Your way of doing it also works fine I guess :)